mbed_targets/mbed_targets.py

```diff
@@ -53,22 +52,9 @@ class MbedTargets:
 
     def __init__(self) -> None:
         """Retrieve target data from the online database."""
-        self._target_data: Optional[
-            List[dict]] = target_database.get_target_data()
+        self._target_data: List[dict] = target_database.get_target_data()
 
-    def __iter__(self) -> 'MbedTargets':
-        """Return the iterator object itself."""
-        # Create a new iterator
-        self._target_iterator: Optional[Iterator] = iter(
-            self._target_data) if self._target_data else None
-        return self
-
-    def __next__(self) -> Optional[MbedTarget]:
-        """Return the next item from the container.
-
-        Returns:
-            An instance of a Mbed target database entry.
-        """
-        # Iterate over the target data and return a new MbedTarget instance for each database entry
-        return MbedTarget(
-            next(self._target_iterator)) if self._target_iterator else None
+    def __iter__(self) -> Iterator["MbedTarget"]:
+        """Yield an MbedTarget on each iteration."""
+        for tgt in self._target_data:
```

Comment:
I would prefer to be more verbose and not have to work out what tgt was shortened from.
```suggestion
        for target in self._target_data:
```